### PR TITLE
feat: add ExecutionPayload & ExecutionPayloadHeader & Withdrawal types

### LINKED ISSUE #55: Add SSZ container: `ExecutionPayload`
Follow these instructions: https://github.com/lambdaclass/lambda_ethereum_consensus/blob/main/native/ssz_nif/README.md

#### `ExecutionPayload`

```python
class ExecutionPayload(Container):
    # Execution block header fields
    parent_hash: Hash32
    fee_recipient: ExecutionAddress  # 'beneficiary' in the yellow paper
    state_root: Bytes32
    receipts_root: Bytes32
    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
    prev_randao: Bytes32  # 'difficulty' in the yellow paper
    block_number: uint64  # 'number' in the yellow paper
    gas_limit: uint64
    gas_used: uint64
    timestamp: uint64
    extra_data: ByteList[MAX_EXTRA_DATA_BYTES]
    base_fee_per_gas: uint256
    # Extra payload fields
    block_hash: Hash32  # Hash of execution block
    transactions: List[Transaction, MAX_TRANSACTIONS_PER_PAYLOAD]
    withdrawals: List[Withdrawal, MAX_WITHDRAWALS_PER_PAYLOAD]  # [New in Capella]
```

### PR DESCRIPTION
**Pull request by phklive**
**Motivation**

adding ExecutionPayload & ExecutionPayloadHeader & Withdrawal types for ssz serialization and deserialization

**Description**
Adding ssz container for ExecutionPayload & ExecutionPayloadHeader & Withdrawal

Closes #55, #57, #56

### CODE REVIEW

File: lib/ssz_types/execution_payload.ex [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          prev_randao: SszTypes.bytes32(),
+          block_number: SszTypes.uint64(),
+          gas_limit: SszTypes.uint64(),
+          gas_used: SszTypes.uint64(),
+          timestamp: SszTypes.uint64(),
+          extra_data: binary(),
+          base_fee_per_gas: binary(),
+          block_hash: SszTypes.hash32(),
+          transactions: list(SszTypes.transaction()),
+          withdrawals: list(SszTypes.Withdrawal)
```
[the remark this answers is not in the document]
Comment (phklive) [reply lib/ssz_types/execution_payload.ex:43]: Indeed

File: lib/ssz_types/execution_payload_header.ex [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          state_root: SszTypes.root(),
+          receipts_root: SszTypes.root(),
+          logs_bloom: binary(),
+          prev_randao: SszTypes.bytes32(),
+          block_number: SszTypes.uint64(),
+          gas_limit: SszTypes.uint64(),
+          gas_used: SszTypes.uint64(),
+          timestamp: SszTypes.uint64(),
+          extra_data: binary(),
+          base_fee_per_gas: binary(),
```
[the remark this answers is not in the document]
Comment (phklive) [reply lib/ssz_types/execution_payload_header.ex:40]: It is supposed to be a u256, added the type as follows:
```
defmodule SszTypes do
  @moduledoc """
  Lists some types used in SSZ structs.
  """

  # Primitive types
  ## Integer types
  @type uint64 :: 0..unquote(2 ** 64 - 1)
  @type uint256 :: 0..unquote(2 ** 256 - 1)
```

File: test/spec/utils/utils.ex [hunk 3]
```diff
@@ -10,7 +10,21 @@ defmodule SpecTestUtils do
   end
 
   def parse_yaml(list) when is_list(list), do: Enum.map(list, &parse_yaml/1)
+  def parse_yaml({"extra_data", x}), do: {:extra_data, parse_as_string(x)}
+  def parse_yaml({"transactions", list}), do: {:transactions, Enum.map(list, &parse_as_string/1)}
   def parse_yaml({k, v}), do: {String.to_atom(k), parse_yaml(v)}
   def parse_yaml("0x" <> hash), do: Base.decode16!(hash, [{:case, :lower}])
+  def parse_yaml("0x"), do: ""
```
Comment (mpaulucci) [inline test/spec/utils/utils.ex:17]: Maybe `<<>>` is more readable? I had to open a console to see that it was the same as "" in elixir xD

Comment (MegaRedHand) [reply test/spec/utils/utils.ex]: This overload isn't actually needed, because `"0x" <> hash` matches `"0x"` 😅

**Pull request merged**